### PR TITLE
[infra] Cleanup overall API

### PR DESCRIPTION
**Pull request by neeldug**
Cleanup based loosely on #10

### CODE REVIEW

**Review by Razvan00Rusu — APPROVED** [review]
All changes look good to me. Do they have any impact on the data races discussed regarding decisions?

File: pkg/infra/main.go [hunk 1]
```diff
@@ -90,15 +89,14 @@ func gameLoop(globalState state.State, agentMap map[uint]agent.Agent, decisionCh
...
+func initialise() (map[uint]agent.Agent, state.State, config.GameConfig) {
 	agentMap := make(map[uint]agent.Agent)
 
 	agentStateMap := make(map[uint]state.AgentState)
-	decisionChannels := make(map[uint]<-chan decision.Decision)
 
 	err := godotenv.Load()
 	if err != nil {
-		logging.Log.Fatalln("No .env file located, using defaults")
+		logging.Log.Infoln("No .env file located, using defaults")
```
Comment (0x6770) [inline pkg/infra/main.go:99]: Maybe use Warn here to bring more attention?

**Review by 0x6770 — APPROVED** [review]
LGTM

**Comment by neeldug** [comment]
> All changes look good to me. Do they have any impact on the data races discussed regarding decisions?

Sorta, has majority ended up fixing some of the data-racy stuff, however, we still ought to add the immutable library and stick an adaptor before giving the "State" to agents and instead provide an immutable view of the State with the Maps converted to Immutable counterparts.

There is a really weird bug with the current implementation as the stamina drain effectively causes the entire cohort of agents to consider cowering together which results in them gaining more health than the monster deals, resulting in an infinite loop of healing more damage than that which is death by the monster.

**Review by Razvan00Rusu — APPROVED** [review]
> There is a really weird bug with the current implementation as the stamina drain effectively causes the entire cohort of agents to consider cowering together which results in them gaining more health than the monster deals, resulting in an infinite loop of healing more damage than that which is death by the monster.

Sounds like a feature not a bug to me 😎

**Review by 0x6770 — APPROVED** [review]
LGTM!

**Comment by neeldug** [comment]
> > There is a really weird bug with the current implementation as the stamina drain effectively causes the entire cohort of agents to consider cowering together which results in them gaining more health than the monster deals, resulting in an infinite loop of healing more damage than that which is death by the monster.
> 
> Sounds like a feature not a bug to me 😎 LGTM

It ends up cycling between these quite often, something to definitely look into, since it's not a race related bug and instead some game mechanic bug with the split damage turning out to be tiny.

Screenshot 2022-11-11 at 00 07 22
Screenshot 2022-11-11 at 00 07 32

**Pull request merged**